lib/parser/grammar.h

```diff
@@ -959,9 +960,16 @@ TYPE_PARSER(construct<BOZLiteralConstant>(BOZLiteral{}))
 // R1124 do-variable -> scalar-int-variable-name
 constexpr auto doVariable{scalar(integer(name))};
 
+// NOTE: In loop-control we allow REAL name and bounds too.
```

Comment:
Rather than use `is_same_v` tests in the template function body, it would be perhaps more clear to just define two overloads (if that's all that are used), or have one general template and one specialization.

Reply:
Good idea, that is cleaner.